File: examples/docs-code-snippets/src/main/java/examples/java/interactions/CancelOrderImpl.java
package examples.java.interactions;

import java.util.List;

public class CancelOrderImpl implements CancelOrder {
    @Override
    public OrderCancelled apply(String orderId, List<String> unavailableProductIds) {
        System.out.printf("Canceling order %s. The following products are unavailable: %s", orderId, unavailableProductIds);
        return new OrderCancelled();
    }
}

File: examples/docs-code-snippets/src/main/kotlin/examples/kotlin/interactions/CancelOrderImpl.kt
package examples.kotlin.interactions

object CancelOrderImpl : CancelOrder {
    override fun apply(orderId: String, unavailableProductIds: List<String>): CancelOrder.OrderCancelled {
        println("Canceling order $orderId. The following products are unavailable: $unavailableProductIds")
        return CancelOrder.OrderCancelled
    }
}